python/ray/data/_internal/compute.py

```diff
@@ -205,6 +205,7 @@ def __init__(
         min_size: int = 1,
         max_size: Optional[int] = None,
         max_tasks_in_flight_per_actor: Optional[int] = None,
+        size: Optional[int] = None,
```

Comment:
It's an incidental API fix.

Reply:
ok, so then we have different behavior between:
* `ActorPoolStrategy(n)`: a pool has minimal size of n
* `ActorPoolStrategy(size=n)`: a pool with fixed size of n

Not sure if these two different behavior will confuse users or not. At least this looks very subtle to me. But I am fine to add it. It may just be an issue for me.

Reply:
Agree with @c21 here, it is a very subtle difference.

Ideally we would have `size` as the first arg so there is no difference. But this does break backwards compatibility...we can log a warning that this behavior has changed.

Reply:
Hmm if backwards compatibility were not a concern, I'd definitely vote for `ActorPoolStrategy(n)` resulting in a fixed size pool of size `n`, and require kwargs specification of all other args (including `min_size` and `max_size`). With backwards compatibility being a concern, a `size` kwarg seems fine.

Reply:
IMO backwards compatibility definitely should be a concern, `ActorPoolStrategy(n)` is a valid statement before, so we cannot assume user not use it. `ActorPoolStrategy` is a public API.

So if we add `size` kwarg here, can we call out the difference between `ActorPoolStrategy(n)` and `ActorPoolStrategy(size=n)` in documentation at least? So users are not caught by it?

Reply:
How about this, we force kwargs for all three of these args, so `ActorPoolStrategy(n)` is not allowed? This does break backwards compatibility but in a simple way.

Let me pull this out into a separate PR and try to do this in a staged way with a warning first.

Reply:
+1